### ISSUE TITLE
Runelite won't use GPU on Linux/Mesa

### ISSUE THREAD

**Issue by genevieve-me** [runelite/runelite#17468]
### Type

GPU

### Operating System

Linux

### OS Version

Linux 6.1.76

### Bug description

I'm on Linux 6.1.76 (NixOS) with Mesa 23.3.3 and a RX 7600 GPU. I'm launching Runelite with Bolt. I tried installing through my package manager and through flatpak and experienced the same behavior. I also tried giving both Bolt/Runelite all permissions via flatseal.

By default, whenever I turn on either the GPU or 117HD plugin, my CPU utilization goes to 100% and FPS drops to <5. However, GPU usage (measured with `amdgpu_top` is near zero. I tried disabling compute shaders as mentioned in https://github.com/runelite/runelite/issues/16354 but did not experience any changes. This is not a duplicate of that issue, as many users there are reporting high GPU utilization.

If I run RuneLite with hardware acceleration forced to OpenGL and the GPU plugin turned on, I get about 10FPS, still no GPU usage, and I have terrible artifacting as described in https://github.com/runelite/runelite/wiki/Disable-Hardware-Acceleration.

If I force hardware acceleration to be off, I have "acceptable" performance (20FPS), no artifacting, and no GPU usage. My CPU utilization stays at around 50%.

My GPU is set up fine for other system uses: I can run Steam games or Blender using ROCm without any problems. I do not have an iGPU (Ryzen 2600 CPU).

Possibly related: https://github.com/117HD/RLHD/issues/21

### RuneLite version

RuneLite 1.10.21.1, Launcher version 2.6.13

### Logs

[client.log](https://github.com/runelite/runelite/files/14201356/client.log)
[launcher.log](https://github.com/runelite/runelite/files/14201357/launcher.log)

**Comment by Adam-** [comment]
You log file shows you have not turned the GPU plugin on. I would try that and see what the log shows.

**Comment by genevieve-me** [comment]
I do have it turned on - I actually just started runelite from scratch with the GPU plugin, ran for a minute or two, then exited. Unfortunately, my logs in `~/.var/app/net.runelite.RuneLite/.runelite/logs` were last changed about 20 minutes ago - I'm not sure why runelite isn't generating them.

![image](https://github.com/runelite/runelite/assets/140119870/3f9a98ef-8028-4340-b075-3ce70bf626f5)

**Comment by Adam-** [comment]
It can't be on because it logs this at very early startup:
```java
				log.info("Using device: {}", GL43C.glGetString(GL43C.GL_RENDERER));
				log.info("Using driver: {}", GL43C.glGetString(GL43C.GL_VERSION));
```

Maybe your client has stopped logging.

**Comment by genevieve-me** [comment]
Yes, I just tried relaunching runelite a few times, and also taking screenshots, but now it doesn't seem to be outputting any sort of files at all. Strange, since it still has all the flatpak permissions possible; I'm not really sure how to debug that.

**Comment by Adam-** [comment]
Try using the appimage maybe? I don't know much about how flatpak permissions work.

**Comment by Adam-** [comment]
llvmpipe is the llvm software rasterizer. So indeed it is not using your gpu. Can you try the appimage? And also get `glxinfo`

**Comment by aHooder** [comment]
Have you tried running `flatpak update`?

**Comment by genevieve-me** [comment]
> Can you try the appimage? And also get glxinfo

I still see `2024-02-07 23:41:53 GMT [Client] INFO  n.r.client.plugins.gpu.GpuPlugin - Using device: llvmpipe (LLVM 16.0.6, 256 bits)` unfortunately.

If there's a way to actually log in and test with the AppImage if I only have access to a Jagex account, please let me know.

[glxinfo.txt](https://github.com/runelite/runelite/files/14202739/glxinfo.txt)


> Have you tried running flatpak update?

Yes, I already had.

**Comment by Adam-** [comment]
> > Can you try the appimage? And also get glxinfo
> 
> I still see `2024-02-07 23:41:53 GMT [Client] INFO n.r.client.plugins.gpu.GpuPlugin - Using device: llvmpipe (LLVM 16.0.6, 256 bits)` unfortunately.
> 
> If there's a way to actually log in and test with the AppImage if I only have access to a Jagex account, please let me know.
> 
> [glxinfo.txt](https://github.com/runelite/runelite/files/14202739/glxinfo.txt)
> 
> > Have you tried running flatpak update?
> 
> Yes, I already had.

If it still says llvmpipe it doesn't matter, it isn't working. glxinfo appears to not be able to identify your gpu either:
```
OpenGL vendor string: Mesa
OpenGL renderer string: llvmpipe (LLVM 16.0.6, 256 bits)
OpenGL core profile version string: 4.5 (Core Profile) Mesa 23.3.3
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
```

There is about where my knowledge ends with gpu support on linux though. Maybe dig into why Mesa thinks you don't have a gpu?

**Comment by genevieve-me** [comment]
For reference, there's also https://github.com/NixOS/nixpkgs/issues/271739 , but that looks like it produces an error in the logs on enabling the GPU plugins, and I don't see anything like that in my logs. Plus, this is also happening on the RuneLite flatpak.

Just to be sure, I tried native/appimage runelite with `LD_LIBRARY_PATH` pointing to libGL and libXxf86vm, but it was still using `llvmpipe`.

**Comment by genevieve-me** [comment]
> Maybe dig into why Mesa thinks you don't have a gpu?

Fair, that's a good point. I thought it could be kernel related, so I tried with 6.7.3, but glxinfo outputs the same. On the other hand, firefox + other software recognizes it as a RX 7600. I also don't know a ton about GPU handling - I'll try to investigate more and narrow down if it's Mesa related or not.

**Comment by genevieve-me** [comment]
This looks like a bug in nixpkgs/nixos's handling of the mesa/rocm AMD stack. Thanks for the help debugging.
